NBitcoin/Transaction.cs

```diff
@@ -553,7 +553,7 @@ private int GetSpecialSize(uint nSize)
 		#endregion
 	}
 
-	public class TxOut : IBitcoinSerializable, IDestination
+	public class TxOut : IBitcoinSerializable, IDestination, IEquatable<TxOut>

```

Comment:
Don't do this, this will not make expected result for crypto currencies with TxOut with other field (like Elements)